qt/python/mantidqt/project/decoderfactory.py

```diff
@@ -8,21 +8,28 @@
 #
 from __future__ import (absolute_import, division, print_function, unicode_literals)
 
+from mantidqt.usersubwindowfactory import UserSubWindowFactory
+
 
 class DecoderFactory(object):
     decoder_list = set([])
 
     @classmethod
-    def find_decoder(cls, tag):
+    def find_decoder(cls, tag, dict):
         """
         This assumes that tag is present in a decoder and if it cannot find a decoder with the tag it raises ValueError
         :param tag: String; The tag used for saving and thus to be reused for loading that encoder back.
+        :param dict: Whilst this should return a decoder, in the event of a C++ Decoder it needs the dictionary that was
+         used for saving.
         :return: Decoder object; The object of a decoder for the tag that was passed.
         """
         for decoder in cls.decoder_list:
             if tag in decoder.tags:
                 return decoder()
-        raise ValueError("Unable to find decoder for tag: " + tag + " Please add the decoder for: " + tag)
+        object_reinstated = UserSubWindowFactory.Instance().decodeWindow(dict, tag)
```

Comment:
Similarly to the `EncoderFactory` I think we should have a higher-level `decode` function and leave `find_decoder` to return a decoder or raise an exception.